File: Gestion_Angular_Front_end/src/app/classes/avarie-dto.ts
import { ProduitDto } from "./produit-dto";

export class AvarieDto {
  id: number= 0;
  code: string= "";
  nom: string= "";
  date: string="";
  motif: string="";
  produitDto: ProduitDto= new ProduitDto();
}

File: backend/src/main/java/com/odji/spring_back_end/avarie/dto/AvarieDto.java
package com.odji.spring_back_end.avarie.dto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.math.BigDecimal;
import java.time.LocalDate;

@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class AvarieDto {

    private Integer id;

    private BigDecimal quantite;

    private LocalDate date;

    private String motif;

    private ProduitDto produit;
}
